### PR TITLE
fix(network): close stream on timeout

### PR DESCRIPTION
**Pull request by themantre**
## Description

This PR closes the stream once the timeout expires. 
This ensures all streams are closed, even if the receiver didn't close them.

### CODE REVIEW

**Comment by alidevjimmy** [comment]
LGTM

File: network/stream.go [hunk 1]
```diff
@@ -51,41 +53,65 @@ func (s *streamService) handleStream(stream lp2pnetwork.Stream) {
...
 	if err != nil {
-		return LibP2PError{Err: err}
+		return nil, LibP2PError{Err: err}
 	}
 
-	return nil
+	// We need to close the stream once it is read by the receiver.
+	// If, for any reason, the receiver doesn't close the stream, we need to close it after a timeout.
+	go func() {
+		timer := time.NewTimer(s.timeout)
```
Comment (Ja7ad) [inline network/stream.go:95]: Please add `defer timer.Stop()`

File: network/stream.go [hunk 2]
```diff
@@ -51,41 +53,65 @@ func (s *streamService) handleStream(stream lp2pnetwork.Stream) {
...
+			buf := make([]byte, 1)
+			_, _ = stream.Read(buf)
+			closed <- true
+		}()
+
+		select {
+		case <-timer.C:
+			s.logger.Warn("stream timeout", "to", pid)
+			_ = stream.Close()
+
```
Comment (Ja7ad) [inline network/stream.go:109]: Please close channel `close(closed)` here

File: network/stream.go [hunk 3]
```diff
@@ -51,41 +53,65 @@ func (s *streamService) handleStream(stream lp2pnetwork.Stream) {
...
+		case <-closed:
+			s.logger.Debug("stream closed", "to", pid)
+			_ = stream.Close()
+		}
```
Comment (Ja7ad) [inline network/stream.go:113]: Please close channel `closed` here too

File: network/stream.go [hunk 4]
```diff
@@ -51,41 +53,65 @@ func (s *streamService) handleStream(stream lp2pnetwork.Stream) {
...
+		go func() {
+			// We need only one byte to read the EOF.
+			buf := make([]byte, 1)
+			_, _ = stream.Read(buf)
+			closed <- true
+		}()
+
+		select {
+		case <-timer.C:
+			s.logger.Warn("stream timeout", "to", pid)
```
Comment (Ja7ad) [inline network/stream.go:107]: Please update warn message

```
stream closed due to timeout
```

File: network/stream.go (hunk 1 above)
Comment (b00f) [reply network/stream.go:95]: Calling `Stop` is not necessary after Go 1.23, as the garbage collector handles stopped or expired timers. Additionally, timers are synchronized in Go 1.23 and later. Checkout here: https://pkg.go.dev/time#NewTimer

File: network/stream.go (hunk 2 above)
Comment (b00f) [reply network/stream.go:109]: No need. The garbage collector handles channels that are no longer needed.

File: network/stream.go (hunk 3 above)
Comment (b00f) [reply network/stream.go:113]: Same as above.

File: network/stream.go (hunk 4 above)
Comment (b00f) [reply network/stream.go:107]: I prefer shorter log...

**Pull request merged**